### DISCUSSION TITLE
Not being able to build in Release mode after upgrading to Uno 5.2

### DISCUSSION THREAD

**Discussion by WenHong-Lau** [unoplatform/uno#16512] [General]
I've recently updated the Uno Platform extension to ensure compatibility with the latest project templates and Uno 5.2. I ran uno-check to verify everything was up to date before creating a project using the "Uno Platform App" template.

However, upon attempting to build the project, I consistently encounter the following error:
```
The "GenerateAppManifestFromAppx" task failed unexpectedly.
```
Occasionally, I also face another issue stating:
```
Failed linking file resources.
```

I have attached the zipped solution along with the binlog
[UnoApp1.zip](https://github.com/unoplatform/uno/files/15157285/UnoApp1.zip)

**Comment by jeromelaban** [comment]
Thanks for the report. Can you provide the environment and command line you're using to make your build?

**Reply by jeromelaban** [reply]
Note that this issue is related to https://github.com/microsoft/WindowsAppSDK/issues/4008, and that building with `msbuild` instead of `dotnet build` can fix this particular issue.

**Reply by WenHong-Lau** [reply]
@jeromelaban I am building through Visual Studio, which I believe is already using MSBuild. However, I've tried:
```
 msbuild /p:Configuration=Release
``` 
as suggested and I'm still getting the same error. I only started experiencing this issue after upgrading to Uno 5.2. When I rolled back to the older version, it built without any issues.

**Reply by limeniye** [reply]
I've tried the following command:
```PS
msbuild /t:Publish /p:Configuration=Release /p:RuntimeIdentifier=win-x64 /p:Platform=x64
```
I've got an errors with Uno.Resizetizer

[MsBuildWithUnoResizetizer.zip](https://github.com/unoplatform/uno/files/15213787/MsBuildWithUnoResizetizer.zip)

```
C:\Users\mykyt\.nuget\packages\uno.resizetizer\1.4.8\build\Uno.Resizetizer.targets(485,3): error Uno.Resizetizer0000: S
ystem.TypeInitializationException: The type initializer for 'SkiaSharp.SKImageInfo' threw an exception. ---> System.Bad
ImageFormatException: An attempt was made to load a program with an incorrect format. (Exception from HRESULT: 0x800700
0B) [R:\ForDelete\UnoApp77\UnoApp77\UnoApp77.csproj::TargetFramework=net8.0-windows10.0.19041]
C:\Users\mykyt\.nuget\packages\uno.resizetizer\1.4.8\build\Uno.Resizetizer.targets(485,3): error Uno.Resizetizer0000:
  at SkiaSharp.SkiaApi.sk_colortype_get_default_8888() [R:\ForDelete\UnoApp77\UnoApp77\UnoApp77.csproj::TargetFramework
=net8.0-windows10.0.19041]
C:\Users\mykyt\.nuget\packages\uno.resizetizer\1.4.8\build\Uno.Resizetizer.targets(485,3): error Uno.Resizetizer0000:
  at SkiaSharp.SKImageInfo..cctor() [R:\ForDelete\UnoApp77\UnoApp77\UnoApp77.csproj::TargetFramework=net8.0-windows10.0
.19041]
C:\Users\mykyt\.nuget\packages\uno.resizetizer\1.4.8\build\Uno.Resizetizer.targets(485,3): error Uno.Resizetizer0000:
  --- End of inner exception stack trace --- [R:\ForDelete\UnoApp77\UnoApp77\UnoApp77.csproj::TargetFramework=net8.0-wi
ndows10.0.19041]
```

**Reply by jeromelaban** [reply]
Thanks. Can you try deleting this folder: `C:\Users\mykyt\.nuget\packages\uno.resizetizer\1.4.8` and try again?

**Reply by limeniye** [reply]
@jeromelaban 
```
"R:\ForDelete\UnoApp77\UnoApp77.Windows\UnoApp77.Windows.csproj" (Publish target) (1) ->
(XamlPreCompile target) ->
  R:\ForDelete\UnoApp77\UnoApp77.Shared\AppHead.xaml.cs(2,11): error CS0234: The type or namespace name 'Resizetizer' d
oes not exist in the namespace 'Uno' (are you missing an assembly reference?) [R:\ForDelete\UnoApp77\UnoApp77.Windows\U
noApp77.Windows.csproj]
```

**Reply by jeromelaban** [reply]
Make sure to run a full nuget restore again, this generally means the nuget package is not completely available.

**Reply by limeniye** [reply]
@jeromelaban 

If I rebuild the Windows project to restore the nuget packages after deleting the folder, I receive the same errors with Uno.Resizetizer


https://github.com/unoplatform/uno/assets/69314237/c86567c0-d1f8-43bc-95d0-5165238460f2

**Reply by jeromelaban** [reply]
Thanks. This particular error is generally caused by missing assets, as the code failing to compile is only generated when there's an icon file.

**Reply by limeniye** [reply]
@jeromelaban 
I have updated the Uno.Resizetizer to the .dev version and removed all Assets from UnoApp project.
I see an issue with **GenerateSplashAssets**

![image](https://github.com/unoplatform/uno/assets/69314237/e31229d3-506e-4afb-8c9e-17cc5a66127e)

**Reply by jeromelaban** [reply]
This is the same error as the one you posted originally, the resizetizer files are not copied at the right place for some reason. We do not have a repro for this specific issue at this time. If you can provide a binlog we'll be able to help.

**Reply by limeniye** [reply]
@jeromelaban 

```PS
msbuild /t:Publish /p:Configuration=Release /p:RuntimeIdentifier=win-x64 /p:Platform=x64 /bl
```

[msbuild.zip](https://github.com/unoplatform/uno/files/15232769/msbuild.zip)

**Reply by jeromelaban** [reply]
Thanks. The build is working properly for me with this command, yours is missing `/r` and the `TargetFramework`:

```
msbuild /r /t:Publish /p:Configuration=Release /p:RuntimeIdentifier=win-x64 /p:Platform=x64 /p:TargetFramework=net8.0-windows10.0.19041 /bl
```

Are you building on an arm64 device?

**Reply by limeniye** [reply]
@jeromelaban I`m tring to publish the Windows project on x64 device.
I've tried your suggestion but I have the same result.
Could you try this command with the solution attached in [this](https://github.com/unoplatform/uno/discussions/16512#discussioncomment-9320901
) reply?
Or could I try to check this problem from another way?

**Reply by jeromelaban** [reply]
The solution you linked is also building properly for me. Could you check that `C:\Users\mykyt\.nuget\packages\uno.resizetizer\1.5.0-dev.58\build\netstandard2.0\libSkiaSharp.dll` actually exists?

**Reply by limeniye** [reply]
@jeromelaban yes
![image](https://github.com/unoplatform/uno/assets/69314237/1d475850-3b14-4f8b-bc16-0298c633ff4b)

**Reply by jeromelaban** [reply]
Thanks. Can you show the details tab of that window?

**Reply by limeniye** [reply]
@jeromelaban 
![image](https://github.com/unoplatform/uno/assets/69314237/6802ea42-95d1-4130-9e07-a9934fa7229b)


P.S. also the same for 1.4.8 for me

**Reply by jeromelaban** [reply]
Thanks. Just to be sure again. Delete the `C:\Users\mykyt\.nuget\packages\uno.resizetizer\1.5.0-dev.58`, then run the command I provided (the one with the `/r`), provide the binlog again please.

**Reply by limeniye** [reply]
@jeromelaban 
[msbuild.zip](https://github.com/unoplatform/uno/files/15237099/msbuild.zip)

https://github.com/unoplatform/uno/assets/69314237/6e359511-f2f3-44f8-949c-6e1a9ce73f89

 btw the modified time was not changed
![image](https://github.com/unoplatform/uno/assets/69314237/84a4dd82-7287-490b-b4ef-ce1135fc58ac)

**Reply by jeromelaban** [reply]
Thanks. I just noticed, there's an interesting difference with the way the build is run. On my side `PROCESSOR_ARCHITECTURE = AMD64` is found and yours is `PROCESSOR_ARCHITECTURE = x86`, which explains why the build cannot pass. Are you specifying this particular environment variable in your system?

**Reply by limeniye** [reply]
@jeromelaban, Indeed. To be honest, I didn't know that. 
I'm not specifying this particular environment variable.
What could you suggest me in this case?
I've tried to change the command:
```PS
msbuild /r /t:Publish /p:Configuration=Release /p:RuntimeIdentifier=win-x86 /p:Platform=x86 /p:TargetFramework=net8.0-windows10.0.19041 /bl
```
But result is the same. 
I've also tried to remove the nuget folder before execution of this command.

**Reply by jeromelaban** [reply]
No, this is not related to the target platform. In your context, msbuild is running in 32bits mode, it could be because of the surrounding shell. This is quite uncommon, though.

**Reply by jeromelaban** [reply]
@limeniye I missed this in your video, but you're using powershell, and the binlog shows that there's an environment variable called `msbuild`, which is using an incorrect path (it's supposed to end with `amd64`). This means that msbuild is running in 32bits mode and we're not supporting that scenario, we'll be updating our tooling to flag that scenario.

**Reply by limeniye** [reply]
@jeromelaban, thanks.
Now I understood you.
I found another shell on my PC:
![image](https://github.com/unoplatform/uno/assets/69314237/3a326486-feff-4e19-b597-616c82cc7258)

And now it works well!

**Reply by jeromelaban** [reply]
Great, thanks for the update!

**Comment by WenHong-Lau** [comment]
Hi @jeromelaban, I've tried building again with msbuild.

I followed the documentation at https://platform.uno/docs/articles/uno-publishing-windows-packaged-unsigned.html to split up the restore and build process.

 These are the commands I used:
 
```
msbuild /r /t:Restore /p:RestorePackagesConfig=true /p:Configuration=Release /p:Platform=x64

msbuild UnoApp1.sln 
 /p:TargetFramework=net8.0-windows10.0.19041
 /p:Platform=x64
 /p:AppxBundlePlatforms=x64
 /p:PublishUnsignedPackage=true
 /p:Configuration=Release
 /p:UapAppxPackageBuildMode=SideloadOnly
 /p:AppxBundle=Always
 /p:AppxPackageDir="O:\Build\"
 /p:GenerateAppxPackageOnBuild=true
 /p:VersionPrefix=1.0.0
 /bl
```
 
The new error I've encountered:

```
"C:\Users\wenhong.lau\source\workspaces\UnoApp1\UnoApp1.sln" (default target) (1) ->
"C:\Users\wenhong.lau\source\workspaces\UnoApp1\UnoApp1\UnoApp1.csproj" (default target) (2) ->
(_GetPriFilesFromPayload target) ->
  C:\Users\wenhong.lau\.nuget\packages\microsoft.windowsappsdk\1.5.240404000\buildTransitive\MrtCore.PriGen.targets(660,5): error MSB4096: The item "C:\Users\wenhong.lau\.nuget\packages\uno.winui\5.2.161/lib/net7.0-windows10.0.19041.0/Uno.UI.Toolkit.pri" in item list "PackagingOutputs" does not define a value for metadata "ProjectName".  In order to use this metadata, either qualify it by specifying %(PackagingOutputs.ProjectName), or ensure that all items in this list define a value for this metadata. [C:\Users\wenhong.lau\source\workspaces\UnoApp1\UnoApp1\UnoApp1.csproj::TargetFramework=net8.0-windows10.0.19041]
```

**Reply by jeromelaban** [reply]
Thanks. This may be related to https://github.com/unoplatform/uno/pull/17088, but we're not able to reproduce.

Could you provide full repro steps, including how you created the project?

**Reply by WenHong-Lau** [reply]
I initially created the project using the wizard.

I tried recreating the project to reproduce the error, but it seems the original issue did not reappear. However, I encountered these new errors:
```

"C:\Users\wenhong.lau.DATGEL\source\repos\UnoApp1\UnoApp1.sln" (Publish target) (1) ->
"C:\Users\wenhong.lau.DATGEL\source\repos\UnoApp1\UnoApp1\UnoApp1.csproj" (Publish target) (2) ->
(_ComputeAppxPackagePayload target) ->
  C:\Users\wenhong.lau.DATGEL\.nuget\packages\microsoft.windowsappsdk\1.5.240404000\buildTransitive\Microsoft.Build.Msix.Packaging.targets(151
5,5): error APPX1101: Payload contains two or more files with the same destination path 'UnoApp1.Common.pdb'. Source files:  [C:\U
sers\wenhong.lau.DATGEL\source\repos\UnoApp1\UnoApp1\UnoApp1.csproj::TargetFramework=net8.0-windows10.0.19
041]
C:\Users\wenhong.lau.DATGEL\.nuget\packages\microsoft.windowsappsdk\1.5.240404000\buildTransitive\Microsoft.Build.Msix.Packaging.targets(1515,
5): error APPX1101: C:\Users\wenhong.lau.DATGEL\source\repos\UnoApp1\UnoApp1.Common\bin\x64\Release\net8.0-windows10.0
.19041\UnoApp1.Common.pdb [C:\Users\wenhong.lau.DATGEL\source\repos\UnoApp1\UnoApp1\UnoApp1.cs
proj::TargetFramework=net8.0-windows10.0.19041]
C:\Users\wenhong.lau.DATGEL\.nuget\packages\microsoft.windowsappsdk\1.5.240404000\buildTransitive\Microsoft.Build.Msix.Packaging.targets(1515,
5): error APPX1101: C:\Users\wenhong.lau.DATGEL\source\repos\UnoApp1\UnoApp1.Common\bin\x64\Release\net8.0-windows10.0
.19041\win-x64\UnoApp1.Common.pdb [C:\Users\wenhong.lau.DATGEL\source\repos\UnoApp1\UnoApp1\UnoApp1.csproj::TargetFramework=net8.0-windows10.0.19041]
  C:\Users\wenhong.lau.DATGEL\.nuget\packages\microsoft.windowsappsdk\1.5.240404000\buildTransitive\Microsoft.Build.Msix.Packaging.targets(151
5,5): error APPX1101: Payload contains two or more files with the same destination path 'UnoApp1.MauiControls.pdb'. Source files:
 [C:\Users\wenhong.lau.DATGEL\source\repos\UnoApp1\UnoApp1\UnoApp1.csproj::TargetFramework=net8.0-windows1
0.0.19041]
C:\Users\wenhong.lau.DATGEL\.nuget\packages\microsoft.windowsappsdk\1.5.240404000\buildTransitive\Microsoft.Build.Msix.Packaging.targets(1515,
5): error APPX1101: C:\Users\wenhong.lau.DATGEL\source\repos\UnoApp1\UnoApp1.MauiControls\bin\x64\Release\net8.0-windo
ws10.0.19041\UnoApp1.MauiControls.pdb [C:\Users\wenhong.lau.DATGEL\source\repos\UnoApp1\UnoApp1\UnoApp1.csproj::TargetFramework=net8.0-windows10.0.19041]-
C:\Users\wenhong.lau.DATGEL\.nuget\packages\microsoft.windowsappsdk\1.5.240404000\buildTransitive\Microsoft.Build.Msix.Packaging.targets(1515,
5): error APPX1101: C:\Users\wenhong.lau.DATGEL\source\repos\UnoApp1\UnoApp1.MauiControls\bin\x64\Release\net8.0-windo
ws10.0.19041\win-x64\UnoApp1.MauiControls.pdb [C:\Users\wenhong.lau.DATGEL\source\repos\UnoApp1\UnoApp1\UnoApp1.csproj::TargetFramework=net8.0-windows10.0.19041]
```

**Reply by jeromelaban** [reply]
Thanks. If you can provide a repro as [well as binlogs](https://platform.uno/docs/articles/uno-builds-troubleshooting.html), this will help.

**Reply by jeromelaban** [reply]
> Here's the repro. I don't see the binlog getting generated

Thanks for the repro. You have added projects that are not using the uno.sdk but are using uno-specific target frameworks. Make sure to use the uno library templates to create thos projects.

**Reply by WenHong-Lau** [reply]
Hi @jeromelaban,
I am using libraries that target .NET Standard 2.0, shared across multiple solutions. If I remove the other target frameworks and leave only .NET Standard 2.0, I encounter the following errors:

```
"C:\Users\wenhong.lau.DATGEL\source\repos\UnoApp1\UnoApp1.sln" (default target) (1) ->
"C:\Users\wenhong.lau.DATGEL\source\repos\UnoApp1\UnoApp1\UnoApp1.csproj" (default target) (2) ->
(_ComputeAppxPackagePayload target) ->
  C:\Users\wenhong.lau.DATGEL\.nuget\packages\microsoft.windowsappsdk\1.5.240404000\buildTransitive\Microsoft.Build.Msi
x.Packaging.targets(1515,5): error APPX1101: Payload contains two or more files with the same destination path 'UnoApp1
.Common.pdb'. Source files:  [C:\Users\wenhong.lau.DATGEL\source\repos\UnoApp1\UnoApp1\UnoApp1.csproj::TargetFramework=
net8.0-windows10.0.19041]
C:\Users\wenhong.lau.DATGEL\.nuget\packages\microsoft.windowsappsdk\1.5.240404000\buildTransitive\Microsoft.Build.Msix.
Packaging.targets(1515,5): error APPX1101: C:\Users\wenhong.lau.DATGEL\source\repos\UnoApp1\UnoApp1.Common\bin\x64\Rele
ase\net8.0-windows10.0.19041\UnoApp1.Common.pdb [C:\Users\wenhong.lau.DATGEL\source\repos\UnoApp1\UnoApp1\UnoApp1.cspro
j::TargetFramework=net8.0-windows10.0.19041]
C:\Users\wenhong.lau.DATGEL\.nuget\packages\microsoft.windowsappsdk\1.5.240404000\buildTransitive\Microsoft.Build.Msix.
Packaging.targets(1515,5): error APPX1101: C:\Users\wenhong.lau.DATGEL\source\repos\UnoApp1\UnoApp1.Common\bin\x64\Rele
ase\net8.0-windows10.0.19041\win-x64\UnoApp1.Common.pdb [C:\Users\wenhong.lau.DATGEL\source\repos\UnoApp1\UnoApp1\UnoAp
p1.csproj::TargetFramework=net8.0-windows10.0.19041]
  C:\Users\wenhong.lau.DATGEL\.nuget\packages\microsoft.windowsappsdk\1.5.240404000\buildTransitive\Microsoft.Build.Msi
x.Packaging.targets(1515,5): error APPX1101: Payload contains two or more files with the same destination path 'UnoApp1
.MauiControls.pdb'. Source files:  [C:\Users\wenhong.lau.DATGEL\source\repos\UnoApp1\UnoApp1\UnoApp1.csproj::TargetFram
ework=net8.0-windows10.0.19041]
C:\Users\wenhong.lau.DATGEL\.nuget\packages\microsoft.windowsappsdk\1.5.240404000\buildTransitive\Microsoft.Build.Msix.
Packaging.targets(1515,5): error APPX1101: C:\Users\wenhong.lau.DATGEL\source\repos\UnoApp1\UnoApp1.MauiControls\bin\x6
4\Release\net8.0-windows10.0.19041\UnoApp1.MauiControls.pdb [C:\Users\wenhong.lau.DATGEL\source\repos\UnoApp1\UnoApp1\U
noApp1.csproj::TargetFramework=net8.0-windows10.0.19041]
C:\Users\wenhong.lau.DATGEL\.nuget\packages\microsoft.windowsappsdk\1.5.240404000\buildTransitive\Microsoft.Build.Msix.
Packaging.targets(1515,5): error APPX1101: C:\Users\wenhong.lau.DATGEL\source\repos\UnoApp1\UnoApp1.MauiControls\bin\x6
4\Release\net8.0-windows10.0.19041\win-x64\UnoApp1.MauiControls.pdb [C:\Users\wenhong.lau.DATGEL\source\repos\UnoApp1\U
noApp1\UnoApp1.csproj::TargetFramework=net8.0-windows10.0.19041]


"C:\Users\wenhong.lau.DATGEL\source\repos\UnoApp1\UnoApp1.sln" (default target) (1) ->
"C:\Users\wenhong.lau.DATGEL\source\repos\UnoApp1\StandardProjects\StandardTest.Common\StandardTest.Common.csproj" (def
ault target) (4:50) ->
(ResolvePackageAssets target) ->
  C:\Program Files\dotnet\sdk\8.0.300\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets(266,
5): error NETSDK1005: Assets file 'C:\Users\wenhong.lau.DATGEL\source\repos\UnoApp1\StandardProjects\StandardTest.Commo
n\obj\project.assets.json' doesn't have a target for 'net8.0-windows10.0.19041'. Ensure that restore has run and that y
ou have included 'net8.0-windows10.0.19041' in the TargetFrameworks for your project. [C:\Users\wenhong.lau.DATGEL\sour
ce\repos\UnoApp1\StandardProjects\StandardTest.Common\StandardTest.Common.csproj::TargetFramework=net8.0-windows10.0.19
041]


"C:\Users\wenhong.lau.DATGEL\source\repos\UnoApp1\UnoApp1.sln" (default target) (1) ->
"C:\Users\wenhong.lau.DATGEL\source\repos\UnoApp1\StandardProjects\StandardTest.Core\StandardTest.Core.csproj" (default
 target) (5:40) ->
  C:\Program Files\dotnet\sdk\8.0.300\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets(266,
5): error NETSDK1005: Assets file 'C:\Users\wenhong.lau.DATGEL\source\repos\UnoApp1\StandardProjects\StandardTest.Core\
obj\project.assets.json' doesn't have a target for 'net8.0-windows10.0.19041'. Ensure that restore has run and that you
 have included 'net8.0-windows10.0.19041' in the TargetFrameworks for your project. [C:\Users\wenhong.lau.DATGEL\source
\repos\UnoApp1\StandardProjects\StandardTest.Core\StandardTest.Core.csproj::TargetFramework=net8.0-windows10.0.19041]


"C:\Users\wenhong.lau.DATGEL\source\repos\UnoApp1\UnoApp1.sln" (default target) (1) ->
"C:\Users\wenhong.lau.DATGEL\source\repos\UnoApp1\StandardProjects\StandardTest.Interop.V3.1\StandardTest.Interop.V3.1.
csproj" (default target) (7:45) ->
  C:\Program Files\dotnet\sdk\8.0.300\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets(266,
5): error NETSDK1005: Assets file 'C:\Users\wenhong.lau.DATGEL\source\repos\UnoApp1\StandardProjects\StandardTest.Inter
op.V3.1\obj\project.assets.json' doesn't have a target for 'net8.0-windows10.0.19041'. Ensure that restore has run and
that you have included 'net8.0-windows10.0.19041' in the TargetFrameworks for your project. [C:\Users\wenhong.lau.DATGE
L\source\repos\UnoApp1\StandardProjects\StandardTest.Interop.V3.1\StandardTest.Interop.V3.1.csproj::TargetFramework=net
8.0-windows10.0.19041]
```

**Reply by jeromelaban** [reply]
Thanks. Can you provide the updated repro that leads to this error?

**Reply by WenHong-Lau** [reply]
same repro just removed the uno target frameworks

**Reply by jeromelaban** [reply]
Thanks. So the error has multiple parts, but the origin of the issue is caused by MAUI embedding on Windows. Let's track in https://github.com/unoplatform/uno/issues/17205.

**Comment by WenHong-Lau** [comment]
Here's the repro. I don't see the binlog getting generated
[UnoApp1.zip](https://github.com/user-attachments/files/15831319/UnoApp1.zip)
